WikipediaApi.py

```diff
@@ -47,9 +47,20 @@ def get_text_and_links(self, title):
         links = response.find_all('a')
         unique_links = set([link["href"].split("/")[-1].split("#")[0]
                             for link in links
-                            if link.parent.name == "p" and link.class_ != "new"])
+                            if link.parent.name == "p" and link.class_ != "new"])  # ignore links to nonexistent pages
         return response.text, unique_links
 
+    def get_canonical_name(self, title):
```

Comment:
Aren't Wikipedia article titles case-insensitive? Would it be safe/simpler to simply convert everything to lowercase?